### PR TITLE
problem 2 answers

### PR DESCRIPTION
**Pull request by S9S99**

### CODE REVIEW

File: answers/src/answers/Problem2.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        }
+        System.out.println(sum);
+    }
+
+    public static long fibonacci(int n){
+        if (n < 2){
+            return n;
+        } else {
+            return fibonacci(n-1) + fibonacci(n-2);
+        }
```
Comment (hatashiro) [inline answers/src/answers/Problem2.java]: Actually, we don't need `else` here.

```java
if (n < 2) {
  return n;
}
return fibonacci(n-1) + fibonacci(n-2);
```

File: answers/src/answers/Problem2.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            if (fibResult % 2 == 0){
+                sum += fibResult;
+            }
+            i++;
+        }
+        System.out.println(sum);
+    }
+
+    public static long fibonacci(int n){
+        if (n < 2){
```
Comment (hatashiro) [inline answers/src/answers/Problem2.java]: Add a space between `)` and `{`

File: answers/src/answers/Problem2.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
+ */
+public class Problem2 {
+    public static void main(String[] args){
+        long sum = 0;
+        int i = 1;
+        long end = 4000000;
+        long fibResult = 1;
+        while (fibResult < end) {
+            fibResult = fibonacci(i);
```
Comment (noraesae) [inline answers/src/answers/Problem2.java]: Why not declaring `fibResult` here?

```java
while (...) {
   long fibResult = fibonacci(i);
   ...
}

File: answers/src/answers/Problem2.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...
+ * By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
+ */
+public class Problem2 {
+    public static void main(String[] args){
+        long sum = 0;
+        int i = 1;
+        long end = 4000000;
+        long fibResult = 1;
+        while (fibResult < end) {
```
Comment (hatashiro) [inline answers/src/answers/Problem2.java]: Inline 4000000 here, or use `final`.

**Review by hatashiro — CHANGES_REQUESTED** [review]

File: answers/src/answers/Problem2.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * Each new term in the Fibonacci sequence is generated
+ * by adding the previous two terms.
+ * By starting with 1 and 2, the first 10 terms will be:
+ * 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...
+ * By considering the terms in the Fibonacci sequence
+ * whose values do not exceed four million,
+ * find the sum of the even-valued terms.
+ */
+public class Problem2 {
+    public static void main(String[] args){
```
Comment (hatashiro) [inline answers/src/answers/Problem2.java]: `){` -> `) {`

File: answers/src/answers/Problem2.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        long sum = 0;
+        int i = 1;
+        long fibResult = 1;
+        while (fibResult < 4000000) {
+            fibResult = fibonacci(i);
+            if (fibResult % 2 == 0){
```
Comment (hatashiro) [inline answers/src/answers/Problem2.java]: `){` -> `) {`

File: answers/src/answers/Problem2.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+                sum += fibResult;
+            }
+            i++;
+        }
+        System.out.println(sum);
+    }
+
+    public static long fibonacci(int n){
```
Comment (hatashiro) [inline answers/src/answers/Problem2.java]: `){` -> `) {`

**Review by hatashiro — CHANGES_REQUESTED** [review]

**Pull request merged**